src/RuleRightCollection.php

```diff
@@ -232,23 +230,21 @@ public function prepareInputDataForProcess($input, $params)
         $params_lower = array_change_key_case($params, CASE_LOWER);
 
        //common parameters
-        $rule_parameters = [
+        $rule_parameters = array_merge($params_lower, [
             'TYPE'       => $params_lower["type"] ?? "",
             'LOGIN'      => $params_lower["login"] ?? "",
             'MAIL_EMAIL' => $params_lower["email"] ?? $params_lower["mail_email"] ?? "",
-            '_groups_id' => $groups
```

Comment:
This part made no sense to me. When `$input` is an array, which it is always expected to be, the input is assigned to a `$groups` variable and added as the `_groups_id` rule parameter. I assume there was some weird behavior in the very early GLPI versions which remains today.

Reply:
This still doesn't make any sense. Why is the entire `$input` array being assigned to `$groups` and then overriding the provided `_groups_id` parameter? Surely my PR is correct to simply use the provided `$params` array and merge the old uppercase params to handle backwards-compatibility.
The current state of the PR with the example rule you provided works properly for me when testing all rules. I'm doubtful there is ever going to be a difference between `$input` and `$params` with this collection because `$use_output_rule_process_as_next_input` remains the default of false.

Reply:
> Why is the entire `$input` array being assigned to `$groups` and then overriding the provided `_groups_id` parameter?

Because all calls to `RuleRightCollection::processAllRules()` are passing a group list directly. It is probably a bad design, but it has been done so far. For instance: https://github.com/glpi-project/glpi/blob/12acbe6ef46074cbe0af81fed1b34cf0cbb76b0e/src/Auth.php#L486-L497


> The current state of the PR with the example rule you provided works properly for me when testing all rules.

I tested it by authenticating myself on GLPI. In this case, when this line is removed, the group criteria does not seems to work anymore.

Reply:
Does this work for you as well?
```php
'_groups_id' => !empty($params_lower['_groups_id']) ? $params_lower['_groups_id'] : $groups,
```
I think that will handle what seems like the proper syntax from the tests and the legacy input from the actual rule processing.